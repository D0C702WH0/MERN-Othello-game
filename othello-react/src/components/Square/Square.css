

.square {
  width: fit-content;
  height: fit-content;
  background: lightgreen;
  border: 1px solid black;
  display: flex;
  align-items: center;
}

.zoom:focus {
  box-shadow: 5px 5px 5px lightslategray;
  outline: none;
  transform: scale(1.2);
}

@media only screen and (min-width:768px) {
  .square {
    width: 3.5rem;
    height: 3.5rem;
  }
}

@media only screen and (min-width:0px) and (max-width:767px) {
  .square {
    width: 2rem;
    height: 2rem;
  }
}