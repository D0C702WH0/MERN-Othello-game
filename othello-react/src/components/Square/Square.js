import React from 'react'
import Pawn from '../Pawn/Pawn'
import PropTypes from 'prop-types'

import './Square.css'

export default function Square ({ value, click, rowI, colI }) {
  /**
   * Allows to return a black, white or blank pawn
   * @param {*} v Value of the pawn passed as props
   */
  const returnPawn = (v) => {
    switch (v) {
      case 'BLACK':
        return (<Pawn color={'black'} />)
      case 'WHITE':
        return (<Pawn color={'white'} />)
      case 'BLANK':
        return (<Pawn />)
      default:
        break
    }
  }

  /**
   * Handle moves on click
   * @param {*} x x square coordinate
   * @param {*} y x square coordinate
   */
  const handleClick = (x, y) => {
    return click(x, y)
  }

  /**
   * Handle moves on keyboard
   * @param {*} e event (used to check if Enter key is pressed)
   * @param {*} x x square coordinate
   * @param {*} y x square coordinate
   */
  const handleEnter = (e, x, y) => {
    const code = e.keyCode || e.charCode
    if (code === 13) {
      return click(x, y)
    }
  }
  return (
    <div
      className="zoom"
      tabIndex={0}
      onKeyPress={(e) => handleEnter(e, rowI, colI)}
      onClick={() => handleClick(rowI, colI)}>
      <div
        tabIndex={-1}
        className="square"
        onKeyPress={(e) => handleEnter(e, rowI, colI)}
        onClick={() => handleClick(rowI, colI)}>
        {returnPawn(value)}
      </div>
    </div>
  )
}

Square.propTypes = {
  value: PropTypes.string,
  click: PropTypes.func,
  rowI: PropTypes.number,
  colI: PropTypes.number
}
