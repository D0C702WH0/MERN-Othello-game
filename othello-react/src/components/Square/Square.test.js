import React from 'react'
import { configure, mount } from 'enzyme'
import TestRenderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'
import Square from './Square'

configure({ adapter: new Adapter() })

let renderer
const click = jest.fn()
beforeEach(() => {
  renderer = mount(
    <Square
      rowI={2}
      colI={3}
      value={'WHITE'}
      click={click} />
  )
})

describe('Square', () => {
  it('Basic rendering', () => {
    const renderer = TestRenderer.create(
      <Square value={'WHITE'}/>)
    const result = renderer.toJSON()
    expect(result).toMatchSnapshot()
  })

  it('should render one pawn', () => {
    expect(renderer.find('.pawn')).toHaveLength(1)
  })

  it('should render one pawn with className "white"', () => {
    expect(renderer.find('.white')).toHaveLength(1)
  })

  it('should render one pawn with className "black" on props value={"BLACK"}', () => {
    renderer.setProps({ value: 'BLACK' })
    expect(renderer.find('.black')).toHaveLength(1)
  })

  // it('Should have "black" as a className', () => {
  //   const instance = renderer.root
  //   const element = instance.findByType('div')
  //   expect(element.props.className.includes('black')).toBe(true)
  // })

  // it('Should change "black" className to white on props change', () => {
  //   renderer.update(<Pawn color={'white'}/>)
  //   const instance = renderer.root
  //   const element = instance.findByType('div')
  //   const result = renderer.toJSON()
  //   expect(result).toMatchSnapshot()
  //   expect(element.props.className.includes('white')).toBe(true)
  // })
})
