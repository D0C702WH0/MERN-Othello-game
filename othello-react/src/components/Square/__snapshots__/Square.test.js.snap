// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square Basic rendering 1`] = `
<div
  className="zoom"
  onClick={[Function]}
  onKeyPress={[Function]}
  tabIndex={0}
>
  <div
    className="square"
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex={-1}
  >
    <div
      className="pawn white"
    />
  </div>
</div>
`;
